euphonic/broadening.py

```diff
@@ -1,14 +1,67 @@
 """
 Functions for broadening spectra
 """
+from typing import List
 import warnings
+
+import numpy as np
+from pint import Unit
 from scipy.optimize import nnls
 from scipy.stats import norm
 from scipy.signal import convolve
-import numpy as np
 
 from euphonic import ureg, Quantity
 
+
+def polynomial_broadening(bins: Quantity,
+                          x: Quantity,
+                          width_polynomial: List[float],
+                          weights: np.ndarray,
+                          width_unit: Unit = None,
+                          width_lower_limit: float = None,
+                          adaptive_error: float = 1e-2) -> Quantity:
+    """Use fast approximate method to apply x-dependent Gaussian broadening
+
+    Typically this is an energy-dependent instrumental resolution function.
+
+    Parameters
+    ----------
+    bins
+        Energy bins for output spectrum
+    x
+        Data positions (to be binned)
+    width_polynomial
+        Coefficients of polynomial fit to energy/width relationship. Order
+        should be consistent with numpy.polyfit / numpy.polyval (i.e. from
+        largest exponent to smallest.) Width is expressed as standard deviation
+        (not FHWM); the conversion factor between these conventions is
+        :math:`FWHM = \sqrt{8 \ln 2} \sigma`
```

Comment:
It's too early for this kind of detail but noting it now so I don't forget, the Numpy docs now recommend the use of [numpy.polynomial](https://numpy.org/doc/stable/reference/routines.polynomials.package.html#module-numpy.polynomial), which unfortunately has the opposite convention with coefficients from [lowest to highest](https://numpy.org/doc/stable/reference/generated/numpy.polynomial.polynomial.Polynomial.html#numpy.polynomial.polynomial.Polynomial)...